projects/online_sequence_classification/clustering/clustering_interface.py

```diff
@@ -0,0 +1,282 @@
+    """
+
+    return self.find_closest_cluster(point)
+
+
+  def prune(self, max_num_clusters):
+    """
+    If there are too many clusters clusters than the max number of clusters 
+    allowed, merge the closest clusters.
+    
+    :param max_num_clusters: (int) max number of clusters allowed.
+    """
+    while len(self.clusters) >= max_num_clusters:
+      self.merge_closest_clusters()
+
+
+  @staticmethod
+  def noisy_sequence(anomaly_score, noisy_anomaly_score=0.3):
+    """
+    Determine whether a temporal sequence is noisy.
+    
+    :param anomaly_score: (float) anomaly score of the temporal memory
+    :param noisy_anomaly_score: (float) threshold to determine whether the 
+      anomaly score is noisy.
+    :return: (bool) whether the sequence is noisy 
+    """
+    if anomaly_score > noisy_anomaly_score:
+      return True
+    else:
+      return False
+
+
+  @staticmethod
+  def stable_sequence(anomaly_score, stable_anomaly_score=0.2):
+    """
+    Determine whether a temporal sequence is stable.
+    
+    :param anomaly_score: (float) anomaly score of the temporal memory
+    :param stable_anomaly_score: (float) threshold to determine whether the 
+      anomaly score is stable.
+    :return: (bool) whether the sequence is stable 
+    """
+    if anomaly_score < stable_anomaly_score:
+      return True
+    else:
+      return False
+
+
+  def average_cluster_distance(self):
+    """
+    Average cluster distance between clusters.
+    
+    :return average_distance: (float) average distance between clusters. 
+    """
+    cluster_distances = []
+    cluster_ids = self.clusters.keys()
+    for i in cluster_ids:
+      for j in cluster_ids:
+        if i != j:
+          ci = self.clusters[i].center.value
+          cj = self.clusters[j].center.value
+          d = self.distance_func(ci, cj)
+          cluster_distances.append(d)
+
+    if len(cluster_distances) > 0:
+      return np.mean(cluster_distances)
+    else:
+      return 0.0
+
+
+  def add_or_merge_cluster(self, cluster, merge_threshold):
+    """
+    Add cluster to the existing clusters or merge it with the closest cluster.
+    
+    :param cluster: (Cluster) the cluster to assign
+    :param merge_threshold: (float) If the distance to the closest 
+      cluster is below the merge threshold, the cluster will be merged with 
+      the closest cluster. Otherwise, if the distance to the closest cluster 
+      is  above the merge threshold, then the cluster will be added to the 
+      existing clusters.
+    """
+    distance_to_closest, closest, = self.find_closest_cluster(cluster.center)
+    if closest and distance_to_closest < merge_threshold:
+      closest.merge(cluster)
+    else:
+      self.add_cluster(cluster)
+
+
+  def create_cluster(self, center=None):
+    """
+    Create a cluster.
+    
+    :param center: (Point) point to add.
+    """
+    cluster_id = len(self.clusters) + 1
+    cluster = Cluster(cluster_id, center)
```

Comment:
It's not the desired behavior, thanks for catching this. I updated the method so that if a center is provided, then it is added to the cluster list of points (the cluster size won't be zero then)